pkg/controller/workload/workload_processor.go

```diff
@@ -682,10 +702,19 @@ func (p *Processor) updateServiceMap(service *workloadapi.Service) error {
 				return nil
 			}
 		}
+
+		// Compare the addresses of the old and new maps to avoid residual.
+		// If the data can be found in the km_service map, it is also stored in the serviceCache.
+		newServiceAddress := getAddressesFromService(service)
+		oldServiceAddress := getAddressesFromService(oldService)
+		removeServiceAddress := nets.CompareIpByte(newServiceAddress, oldServiceAddress)
+		if err := p.deleteOutdataFrontend(removeServiceAddress); err != nil {
```

Comment:
donot call outdated, by this you make the function limited to process outdated address only. But actually it provide general functionality

Reply:
Done